keentools/geotracker/interface/filedialogs.py

```diff
@@ -652,3 +670,109 @@ def execute(self, context):
             _log.error(f'PRECACLC Exception:\n{str(err)}')
             self.report({'ERROR'}, str(err))
         return {'FINISHED'}
+
+
+class GT_OT_TextureFileExport(Operator, ExportHelper):
+    bl_idname = GTConfig.gt_texture_file_export_idname
+    bl_label = buttons[bl_idname].label
+    bl_description = buttons[bl_idname].description
+    bl_options = {'REGISTER', 'INTERNAL'}
+
+    filter_folder: BoolProperty(
+        name='Filter folders',
+        default=True,
+        options={'HIDDEN'},
+    )
+    filter_image: BoolProperty(
+        name='Filter image',
+        default=True,
+        options={'HIDDEN'},
+    )
+
+    file_format: EnumProperty(name='Image file format',
+                              items=_image_format_items(default='PNG',
+                                                        show_exr=True),
+                              description='Choose image file format',
+                              update=_update_format)
+    quality: IntProperty(name='Image quality', default=100, min=0, max=100)
+    use_half_precision: BoolProperty(name='Use half-precission', default=True)
+
+    check_existing: BoolProperty(
+        name='Check Existing',
+        description='Check and warn on overwriting existing files',
+        default=True,
+        options={'HIDDEN'},
+    )
+
+    filename_ext: StringProperty(default='.png')
+
+    filepath: StringProperty(
+        default='baked_tex',
+        subtype='FILE_PATH'
+    )
+
+    def check(self, context):
+        change_ext = False
+
+        filepath = self.filepath
+        sp = os.path.splitext(filepath)
+
+        if sp[1] in {'.jpg', '.', '.png', '.PNG', '.JPG', '.JPEG',
+                     '.jpeg', '.exr', '.EXR'}:
+            filepath = sp[0]
+
+        filepath = ensure_ext(filepath, self.filename_ext)
+
+        if filepath != self.filepath:
+            self.filepath = filepath
+            change_ext = True
+
+        return change_ext
+
+    def draw(self, context):
+        layout = self.layout
+        layout.label(text='Image file format')
+        layout.prop(self, 'file_format', expand=True)
+
+        if self.file_format == 'JPEG':
+            layout.prop(self, 'quality', slider=True)
+        if self.file_format == 'OPEN_EXR':
+            layout.prop(self, 'use_half_precision')
+            layout.prop(bpy_image_settings(), 'exr_codec')
+
+    def execute(self, context):
+        _log.output(f'START SAVE TEXTURE: {self.filepath}')
+        geotracker = get_current_geotracker_item()
+        if not geotracker:
+            return {'CANCELLED'}
+
+        tex = find_bpy_image_by_name(geotracker.preview_texture_name())
+        if tex is None:
+            return {'CANCELLED'}
+        tex.filepath = self.filepath
+        # Blender doesn't change file_format after filepath assigning, so
```

Comment:
Provide more details about this bugfix

Reply:
We don't know why it works )
This solution was found many years ago. May be it forces Blender to update scene depsgraph.
This comment was inherited with old code